libdevcore/picosha2.h

```diff
@@ -115,9 +115,8 @@ inline uint32_t ssig1(uint32_t x)
 }
 
 template<typename RaIter1, typename RaIter2>
-void hash256_block(RaIter1 message_digest, RaIter2 first, RaIter2 last)
+void hash256_block(RaIter1 message_digest, RaIter2 first, RaIter2)
 {
-	(void)last; // FIXME: check this is valid
```

Comment:
The upstream version of this now added an assertion here...
https://github.com/okdshin/PicoSHA2/blob/b699e6c900be6e00152db5a3d123c1db42ea13d0/picosha2.h#L93

Maybe it's generally a good idea to update it? We can consider that separately from this PR, though.

Reply:
Nah I'll update it here

Reply:
Is this really the only change compared to upstream :-)? Do we want to use one of our ``assertThrow``s for the assertion rather than a plain ``assert`` (that will abort the process instead of throwing an exception)?

Reply:
If we use an ``assertThrow`` (with whatever Exception type) for this, then it'd actually no longer need ``[[maybe_unused]]``...

Reply:
We don't use `solAssert` here?

Reply:
@Marenz in `libdevcore` we don't use `solAssert`. I'll replace by `assertThrow`

Reply:
@ekpyron Apart from style that's the only change compared to upstream

Reply:
Ok... just because I saw those ``PICOSHA2_BUFFER_SIZE_FOR_INPUT_ITERATOR`` and ``word_t`` things there I expected more...

Reply:
But yeah, as far as I'm concerned we can live without the assertion here... A new kind of exception for ``assertThrow`` seems overkill... ``assert`` is not so nice, but would be OK I guess... we could just throw a good old ``std::runtime_error``... but none of that's really good...

Reply:
@ekpyron wait, I forgot to look at other parts of the file, there are other differences indeed.
Maybe we should just use the upstream file completely

Reply:
Yeah, sounds good. Not sure it's worth updating it at all.

Reply:
(we may have intentionally modified it for the keccak256/sha256 difference and stuff, I'm not sure, we'd need to look into it)

Reply:
Created https://github.com/ethereum/solidity/issues/8119 to discuss that, this PR can be merged as is